src/sentry/api/endpoints/organization_index.py

```diff
@@ -217,6 +222,10 @@ def post(self, request: Request) -> Response:
         if serializer.is_valid():
             result = serializer.validated_data
 
+            region_name = result.get("region", None)
+            if region_name and region_name != get_local_region().name:
+                return Response({"detail": "Invalid region name provided"}, status=400)
```

Comment:
This error message is UI visible, so we may need to reword this a touch 🤔

Reply:
`"Invalid data storage location provided"` would match our phrasing on forms

Reply:
True, but it also mismatches the parameter name. Maybe I should rename this to "dataStorageLocation" to keep things consistent between the UI and API?